app/controllers/application_controller.rb

```diff
@@ -14,20 +14,78 @@ class ApplicationController < Sinatra::Base
   def serialization(weather_response, token)
     if weather_success?(weather_response)
       forecast = Forecast.new(weather_response)
-      spotify_response = SpotifyService.new.playlist(token, forecast.main_description)
-      playlist = spotify_response[:playlists][:items].shuffle.first
-      if playlist_success?(spotify_response)
-        WeatherMusicSerializer.new(forecast, Playlist.new(playlist)).data_hash.to_json
-      else
-        WeatherMusicSerializer.new.no_playlist_response.to_json
+
+      response = SpotifyService.new.weather_tracks_first_50(token)
+      song_ids = response[:items].map {|item| item[:track][:id]}.shuffle
+      five_ids = song_ids.take(5)
+      seed_tracks = five_ids.join(",")
+
+      result = SpotifyService.new.create_playlist(target_valence(forecast), target_speech(forecast), target_mode(forecast), target_energy(forecast), target_tempo(forecast), seed_tracks, token)
+
+      tracks = result[:tracks].map do |data|
+        Track.new(data)
       end
+      WeatherMusicSerializer.new(forecast, tracks).data_hash.to_json
     else
       WeatherMusicSerializer.new.no_city_response.to_json
     end
   end
 
-  def playlist_success?(spotify_response)
-    spotify_response[:playlists][:items].empty? == false
+  def target_valence(forecast)
+    max = 116.0
+    min = 0.0
+    val = forecast.temp.to_f
```

Comment:
These are sick. I think we still need to mess around with how we're creating our algorithms to get more precise results.